pkg/cluster/internal/providers/docker/provider.go

```diff
@@ -53,6 +53,8 @@ type Provider struct {
 
 // Provision is part of the providers.Provider interface
 func (p *Provider) Provision(status *cli.Status, cfg *config.Cluster) (err error) {
+	cfg.CgroupRoot = "/kubelet"
```

Comment:
if is the default, I think it should go here `pkg/internal/apis/config/default.go`

Reply:
It should only be set for the Docker provider, so I don't think I can set it in there unless there's a good way to check if the Docker provider is in-use.

Reply:
@aojea any thoughts? I'd like to keep this moving.

Reply:
I'm in deep waters here with the cgroups stuff :smile: ,
 I was reviewing to try to keep it as compatible as possible between the different provider, for the final review you need @BenTheElder

Reply:
> It should only be set for the Docker provider, so I don't think I can set it in there unless there's a good way to check if the Docker provider is in-use.

Why wouldn't we set this unconditionally? 
We want things to be very reproducible between CI and local, behavior branching is kept to a minimum.

Reply:
We do unconditionally set it for Docker, but I previously wasn't able to get it working with podman so we disabled it there, per https://github.com/kubernetes-sigs/kind/pull/1747#discussion_r462606918. But, it's been a few weeks, let me try a rebase and have another go at podman.

Reply:
Alright, rebased and got podman working with the cgroup root. So we set "/kubelet" in `default.go` so that it can be propagated to the kubeadm config and the container that gets created. This allows us to avoid hardcoding "/kubelet" in more than one place (the kubeadm configs and the entrypoint script) as well as disable the cgroup root in the kubeadm configuration for legacy images if we have a way to detect them (https://github.com/kubernetes-sigs/kind/pull/1747#discussion_r463273752).

Can you give this another review?

Reply:
@BenTheElder is it possible to get some feedback here?

Reply:
@BenTheElder  what is the status of this PR?